external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaOffsetReader.scala

```diff
@@ -396,6 +406,8 @@ private[kafka010] class KafkaOffsetReader(
 }
 
 private[kafka010] object KafkaOffsetReader {
+  // offsets are not instances of Optional, we need special state for None
+  val EMPTY_OFFSET: Long = -100L
```

Comment:
I'm not sure I follow why this is needed. Normally, if fetchOffsetsByTime sometimes needs to return offsets and sometimes needs to return None, it should just return Option[Offset] values.

Reply:
Completely agree with you (this is what the comment is trying to express). I tried to go with least invasive changes, but since you've pointed this out, I can change the offset map everywhere from:
`Map[TopicPartition, Long]` to `Map[TopicPartition, Option[Long]]`

Reply:
I don't think it needs to be an Option[Long] everywhere. Just until we decide what value we want to pass to Kafka.

Reply:
Changed to Option[Long] where it was necessary